coderd/provisionerjobs_test.go

```diff
@@ -27,162 +27,255 @@ import (
+	// the API response includes both worker_id and worker_name fields
+	t.Run("AssignedProvisionerJob_IncludesWorkerIDAndName", func(t *testing.T) {
 		t.Parallel()
-		ctx := testutil.Context(t, testutil.WaitMedium)
-		jobs, err := templateAdminClient.OrganizationProvisionerJobs(ctx, owner.OrganizationID, &codersdk.OrganizationProvisionerJobsOptions{
-			Limit: 1,
+
+		db, ps := dbtestutil.NewDB(t, dbtestutil.WithDumpOnFailure())
+		client, _, coderdAPI := coderdtest.NewWithAPI(t, &coderdtest.Options{
+			IncludeProvisionerDaemon: false,
+			Database:                 db,
+			Pubsub:                   ps,
 		})
+		provisionerDaemonName := "provisioner_daemon_test"
+		provisionerDaemon := coderdtest.NewTaggedProvisionerDaemon(t, coderdAPI, provisionerDaemonName, map[string]string{"owner": "", "scope": "organization"})
+		owner := coderdtest.CreateFirstUser(t, client)
+		templateAdminClient, _ := coderdtest.CreateAnotherUser(t, client, owner.OrganizationID, rbac.ScopedRoleOrgTemplateAdmin(owner.OrganizationID))
+
+		version := coderdtest.CreateTemplateVersion(t, client, owner.OrganizationID, nil)
+		coderdtest.AwaitTemplateVersionJobCompleted(t, client, version.ID)
+		template := coderdtest.CreateTemplate(t, client, owner.OrganizationID, version.ID)
+
+		workspace := coderdtest.CreateWorkspace(t, client, template.ID)
+		coderdtest.AwaitWorkspaceBuildJobCompleted(t, client, workspace.LatestBuild.ID)
+
+		// Stop the provisioner so it doesn't grab any more jobs
+		err := provisionerDaemon.Close()
 		require.NoError(t, err)
-		require.Len(t, jobs, 1)
-	})
 
-	// For now, this is not allowed even though the member has created a
-	// workspace. Once member-level permissions for jobs are supported
-	// by RBAC, this test should be updated.
-	t.Run("MemberDenied", func(t *testing.T) {
-		t.Parallel()
-		ctx := testutil.Context(t, testutil.WaitMedium)
-		jobs, err := memberClient.OrganizationProvisionerJobs(ctx, owner.OrganizationID, nil)
-		require.Error(t, err)
-		require.Len(t, jobs, 0)
+		t.Run("List_AssignedProvisionerJob_IncludesWorkerIDAndName", func(t *testing.T) {
+			t.Parallel()
+
+			ctx := testutil.Context(t, testutil.WaitMedium)
+
+			// Get provisioner daemon responsible for executing the provisioner jobs
+			provisionerDaemons, err := db.GetProvisionerDaemons(ctx)
+			require.NoError(t, err)
+			require.Equal(t, 1, len(provisionerDaemons))
+			require.Equal(t, provisionerDaemonName, provisionerDaemons[0].Name)
+
+			// Get provisioner jobs
+			jobs, err := templateAdminClient.OrganizationProvisionerJobs(ctx, owner.OrganizationID, nil)
+			require.NoError(t, err)
+			require.Equal(t, 2, len(jobs))
+
+			for _, job := range jobs {
+				require.Equal(t, owner.OrganizationID, job.OrganizationID)
+				require.Equal(t, database.ProvisionerJobStatusSucceeded, database.ProvisionerJobStatus(job.Status))
+
+				// Guarantee that provisioner jobs contain the provisioner daemon ID and name
+				require.Equal(t, provisionerDaemons[0].ID, *job.WorkerID)
```

Comment:
Potential for nil-ptr dereference for `job.WorkerID`? If yes, the resulting test failure would not be ideally helpful. Consider verifying it's non-nil or perhaps the comparison works if you change it to `&provisionerDAemons[0].ID, job.WorkerID` (did not test this). Same in the test below.

Reply:
`testify/assert` has some nice properties you can use here:

```
if assert.NotEmpty(t, provisionerDaemons) {
    require.Equal(...)
}
```

Reply:
Good catch, thanks! I’ve updated the test to compare pointers directly using `require.Equal(t, &provisionerDaemons[0].ID, job.WorkerID)` to avoid the risk of nil-pointer dereference. Also wrapped it in an `assert.NotEmpty` check to ensure the slice isn’t empty. Commit: https://github.com/coder/coder/pull/17877/commits/98f3f4dc584bd0eee375fc8336129a766a369aad